src-plugins/mscExportVideo/mscExportVideo.cpp

```diff
@@ -0,0 +1,283 @@
+#include "mscExportVideo.h"
+
+#include <dtkCore/dtkAbstractProcessFactory.h>
+
+#include <vtkImageCanvasSource2D.h>
+#include <vtkImageData.h>
+#include <vtkJPEGWriter.h>
+#include <vtkOggTheoraWriter.h>
+#include <vtkQImageToImageSource.h>
+#include <vtkSmartPointer.h>
+
+// /////////////////////////////////////////////////////////////////
+// ExportVideoPrivate
+// /////////////////////////////////////////////////////////////////
+namespace msc
+{
+class ExportVideoPrivate
+{
+public:
+    QVector< QImage > imagesArray;
+    int width;
+    int height;
+
+    // User parameters
+    QString filename;
+    int format;            // Cf. VideoFormat in header file
+    int frameRate;         // Frame per second
+    bool subsampling;      // Is the video to be encoded using 4:2:0 subsampling?
+    int quality;           // 0 = low, 1 = medium, 2 = high
+    bool exportOnlyAsJPEG; // Can be used for home encoding
+};
+
+// /////////////////////////////////////////////////////////////////
+// ExportVideo
+// /////////////////////////////////////////////////////////////////
+
+ExportVideo::ExportVideo() : medAbstractProcess(), d(new ExportVideoPrivate)
+{
+    d->imagesArray.clear();
+    d->width  = 0;
+    d->height = 0;
+
+    // User parameters
+    d->format = OGGVORBIS;
+    d->filename = "";
+    d->frameRate = 1;
+    d->subsampling = false;
+    d->quality = 2;
+    d->exportOnlyAsJPEG = false;
+}
+
+ExportVideo::~ExportVideo()
+{
+    delete d;
+}
+
+bool ExportVideo::registered()
+{
+    return dtkAbstractProcessFactory::instance()->registerProcessType("msc::ExportVideo", createExportVideo);
+}
+
+QString ExportVideo::description() const
+{
+    return "ExportVideo";
+}
+
+QString ExportVideo::identifier() const
+{
+    return description();
+}
+
+void ExportVideo::setParameter(int* data)
```

Comment:
Updated function: this receives a int array of every pixel (R, G and B) of the current screenshot